server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/delete/ClusterDeleteWeightedRoutingRequest.java

```diff
@@ -8,37 +8,142 @@
 
 package org.opensearch.action.admin.cluster.shards.routing.weighted.delete;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.OpenSearchGenerationException;
+import org.opensearch.OpenSearchParseException;
 import org.opensearch.action.ActionRequestValidationException;
 import org.opensearch.action.support.clustermanager.ClusterManagerNodeRequest;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.common.bytes.BytesReference;
 import org.opensearch.common.io.stream.StreamInput;
 import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.DeprecationHandler;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.common.xcontent.XContentType;
 
 import java.io.IOException;
+import java.util.Map;
 
 /**
  * Request to delete weights for weighted round-robin shard routing policy.
  *
  * @opensearch.internal
  */
 public class ClusterDeleteWeightedRoutingRequest extends ClusterManagerNodeRequest<ClusterDeleteWeightedRoutingRequest> {
-    public ClusterDeleteWeightedRoutingRequest() {}
+    private static final Logger logger = LogManager.getLogger(ClusterDeleteWeightedRoutingRequest.class);
+
+    private long version;
+    private String awarenessAttribute;
+
+    public void setVersion(long version) {
+        this.version = version;
+    }
+
+    ClusterDeleteWeightedRoutingRequest() {
+        this.version = WeightedRoutingMetadata.VERSION_UNSET_VALUE;
+    }
 
     public ClusterDeleteWeightedRoutingRequest(StreamInput in) throws IOException {
         super(in);
+        version = in.readLong();
+        if (in.available() != 0) {
+            awarenessAttribute = in.readString();
+        }
+    }
```

Comment:
Can we add a check to read these params only if OpenSearch version is 2.5 and above else there is a possibility that mixed version cluster tests will fail with serde issues. We might want to do the same in main as well.

Reply:
Since the apis are marked experimental, do you think we should put effort on this. The change will require effort and testing to make both the flows work. 

Even if we just fix the serialization/deserialization the request will fail eventually. So if we need to support mixed clusters, we have to build backward compatibility